### PR TITLE
Ch90/send webhook when video transitions to ready

### PR DESCRIPTION
**Pull request by AlineRibeiro**

### CODE REVIEW

File: app/jobs/send_video_status_webhook_job.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class SendVideoStatusWebhookJob < ApplicationJob
+  queue_as :default
+
+  def perform(webhook_subscription, video)
```
Comment (guialbuk) [inline app/jobs/send_video_status_webhook_job.rb]: Pass the `WebhookSubscription`and `Video`ids as arguments. This is a best practice to make the job arguments simple and readable.
Sidekiq documentation on that: https://github.com/mperham/sidekiq/wiki/Best-Practices#1-make-your-job-parameters-small-and-simple

File: app/services/split_variant_into_segments.rb [hunk 2]
```diff
@@ -19,7 +22,10 @@ def perform(variant)
 
       File.delete(playlist_path)
 
-      video.process! if video.may_process?
+      if video.may_process?
+        video.process!
+        SendVideoStatusWebhookJob.perform_later(webhook_subscription, video)
```
Comment (guialbuk) [inline app/services/split_variant_into_segments.rb]: You don't have to enqueue if `webhook_subscription` is null

File: app/services/send_video_status_webhook.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class SendVideoStatusWebhook
+  def self.perform(webhook_subscription, video)
+    if webhook_subscription
+      RestClient.post webhook_subscription.url.to_s,
```
Comment (guialbuk) [inline app/services/send_video_status_webhook.rb]: Is `.to_s`needed?

File: app/services/split_variant_into_segments.rb (hunk 2 above)
Comment (guialbuk) [reply app/services/split_variant_into_segments.rb]: The logic of finding the webhook subscription and enqueuing the job should belong to the state machine `proccess`event.

**Review by guialbuk — COMMENTED** [review]
🚀

File: app/services/send_video_status_webhook.rb (hunk 3 above)
Comment (AlineRibeiro) [reply app/services/send_video_status_webhook.rb]: It does not work without it. I dont know why though. I understand the source is a string already; right?

**Pull request merged**